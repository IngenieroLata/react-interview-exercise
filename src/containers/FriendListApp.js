import React, { Component } from 'react';
import styles from './FriendListApp.css';
import { connect } from 'react-redux';

import { addFriend, deleteFriend, starFriend } from '../actions/FriendsActions';
import { chagePage } from '../actions/PageActions';
import { FriendList, AddFriendForm, Pagination } from '../components';

export class FriendListApp extends Component {
  render() {
    const { friendsById, pagination } = this.props;

    const actions = {
      addFriend: this.props.addFriend,
      deleteFriend: this.props.deleteFriend,
      starFriend: this.props.starFriend,
      chagePage: this.props.chagePage
    };

    return (
      <div className={styles.friendListApp}>
        <h1>The FriendList</h1>
        <AddFriendForm addFriend={actions.addFriend} />
        <FriendList friends={friendsById} actions={actions} />
        <Pagination {...pagination} onPageChange={actions.chagePage} />
      </div>
    );
  }
}

function mapStateToProps({ friendlist: { friendsById }, pagination }) {
  const { limit, page } = pagination;
  const offset = limit * page;
  return {
    friendsById: friendsById.slice(offset - limit, offset),
    pagination
  };
}

export default connect(
  mapStateToProps,
  {
    addFriend,
    deleteFriend,
    starFriend,
    chagePage
  }
)(FriendListApp);
