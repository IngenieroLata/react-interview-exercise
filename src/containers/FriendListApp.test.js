import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { FriendListApp } from './FriendListApp';

const props = {
  friendsById: [
    {
      name: 'Theodore Roosevelt',
      starred: true,
      gender: 'male'
    },
    {
      name: 'Abraham Lincoln',
      starred: false,
      gender: 'male'
    },
    {
      name: 'George Washington',
      starred: false,
      gender: 'male'
    }
  ],
  pagination: {
    total: 3,
    limit: 2,
    page: 1
  },
  addFriend: jest.fn(),
  chagePage: jest.fn()
};

describe('<FriendListApp />', () => {
  const wrapper = shallow(<FriendListApp {...props} />);

  it('h1 contains corrext text', () => {
    expect(wrapper.find('h1').text()).toBe('The FriendList');
  });

  it('Matches the snapshot render', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
