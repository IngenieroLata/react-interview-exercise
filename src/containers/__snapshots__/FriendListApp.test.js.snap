exports[`<FriendListApp /> Matches the snapshot render 1`] = `
<div>
  <h1>
    The FriendList
  </h1>
  <AddFriendForm
    addFriend={[Function]} />
  <FriendList
    actions={
      Object {
        "addFriend": [Function],
        "chagePage": [Function],
        "deleteFriend": undefined,
        "starFriend": undefined,
      }
    }
    friends={
      Array [
        Object {
          "gender": "male",
          "name": "Theodore Roosevelt",
          "starred": true,
        },
        Object {
          "gender": "male",
          "name": "Abraham Lincoln",
          "starred": false,
        },
        Object {
          "gender": "male",
          "name": "George Washington",
          "starred": false,
        },
      ]
    } />
  <Pagination
    limit={2}
    onPageChange={[Function]}
    page={1}
    total={3} />
</div>
`;
